source/brailleViewer/brailleViewerGui.py

```diff
@@ -0,0 +1,148 @@
+from typing import List
+
+import wx
+import gui
+import config
+from logHandler import log
+
+BRAILLE_UNICODE_PATTERNS_START = 0x2800
+SPACE_CHARACTER = u" "
+
+
+# Inherit from wx.Frame because these windows show in the alt+tab menu (where miniFrame does not)
+# wx.Dialog causes a crash on destruction when multiple were created at the same time (speechViewer
+# may start at the same time)
+class BrailleViewerFrame(wx.Frame):
+
+	# Translators: The title of the NVDA Braille Viewer tool window.
+	title = _("NVDA Braille Viewer")
+
+	def __init__(self, numCells, onDestroyed):

```

Comment:
It is a bit confusing that this is a parameter to the class constructor. It is only used to create an initial row of empty braille cells. Is it really necessary to keep it?
If so, I'd rather have the main frame calculate the number of braille cells to use.

Reply:
It allows the GUI to initialise to the right size. I think I would prefer to keep it.

Reply:
Doesn't the gui require a reinit if you switch from no display with braille viewer on to a display with 80 cells?